Components/WindowProcess.cs

```diff
@@ -153,21 +164,23 @@ internal static uint GetThreadIDFromWindowHandle(IntPtr hwnd)
         /// </summary>
         /// <param name="pid">The id of the process/param>
         /// <returns>A string representing the process name or an empty string if the function fails</returns>
-        internal static string GetProcessNameFromProcessID(uint pid)
+        internal static (string , string) GetProcessNameAndImageFromProcessID(uint pid)
         {
             var processHandle = NativeMethods.OpenProcess(ProcessAccessFlags.QueryLimitedInformation, true, (int)pid);
-            var processName = new StringBuilder(MaximumFileNameLength);
-
-            if (NativeMethods.GetProcessImageFileName(processHandle, processName, MaximumFileNameLength) != 0)
-            {
-                _ = CloseHandleIfNotNull(processHandle);
-                return processName.ToString().Split('\\').Reverse().ToArray()[0];
-            }
-            else
-            {
-                _ = CloseHandleIfNotNull(processHandle);
-                return string.Empty;
-            }
+            var stringBuilder = new StringBuilder(MaximumFileNameLength);
+            var processName = String.Empty;
+            var processImage = String.Empty;
+            int capacity = MaximumFileNameLength;
+
+            if (NativeMethods.QueryFullProcessImageName(processHandle, 0, stringBuilder, ref capacity))
+                processImage = stringBuilder.ToString();
+
+            stringBuilder = new StringBuilder(MaximumFileNameLength);
+            if (NativeMethods.GetProcessImageFileName(processHandle, stringBuilder, MaximumFileNameLength) != 0)
+                processName = stringBuilder.ToString().Split('\\').Reverse().ToArray()[0];
+            
+            _ = CloseHandleIfNotNull(processHandle);
```

Comment:
Re-using the processHandle here, and closing it if not null, keeping the impact to a minimum